pm_uep_opt_write/pm_uep_opt_write.xml

```diff
@@ -47,32 +47,53 @@
         <param type="data" name="muonated_structures" label="Muonated structures (.zip)" format="zip" help="A zip folder containing muonated structures, as generated by the 'Generate muonated structures' tool or pm-muairss."/>
         <param type="data" name="charge_density" label="Charge density file (.den_fmt)" format="txt" help="The charge density file created by CASTEP during your initial DFT simulation for the structure."/>
         <param type="data" name="castep_log" label="CASTEP log (.castep)" format="txt" help="The CASTEP log for your initial DFT simulation for the structure."/>
+        <param type="hidden" name="testing" label="Test mode" value="false"/>
     </inputs>
     <outputs>
         <data label="UEP results for $muonated_structures.name" name="uep_results" format="zip" from_work_dir="out_zip.zip"/>
-        <data label="File tree (testing only)" name="file_tree" format="txt" from_work_dir="tree.txt" hidden="true"/>
+        <data label="File tree (testing only)" name="file_tree" format="txt" from_work_dir="tree.txt" hidden="true">
+            <filter>(testing == "true")</filter>
+        </data>
     </outputs>
     <tests>
         <test>
+            <param name="testing" value="true"/>
             <param name="muonated_structures" value="muonated.zip" ftype="zip"/>
             <param name="charge_density" value="Si.den_fmt" ftype="txt"/>
             <param name="castep_log" value="Si.castep" ftype="txt"/>
-            <output name="file_tree" file="tree.txt" ftype="txt" compare="sim_size" delta="10"/>
+            <output name="file_tree" file="tree.txt" ftype="txt" sort="true">
+                <assert_contents>
+                    <has_size value="1780"/>
+                </assert_contents>
+            </output>
         </test>
         <test>
+            <param name="testing" value="true"/>
             <param name="muonated_structures" value="muonated-default-folder.zip" ftype="zip"/>
             <param name="charge_density" value="Si.den_fmt" ftype="txt"/>
             <param name="castep_log" value="Si.castep" ftype="txt"/>
-            <output name="file_tree" file="tree-default-folder.txt" compare="sim_size" ftype="txt" delta="10"/>
+            <output name="file_tree" file="tree-default-folder.txt" sort="true">
```

Comment:
If we are comparing the file directly here, why do we then have another assert on size?

Reply:
Good catch, will remove